docs/tasks/configure-pod-container/configure-liveness-readiness-probes.md

```diff
@@ -0,0 +1,257 @@
+---
+redirect_from:
+- "/docs/user-guide/liveness/"
+- "/docs/user-guide.liveness.html"
+title: Configuring Liveness and Readiness Probes
+---
+
+{% capture overview %}
+
+This page shows how to configure liveness and readiness probes for Containers.
+
+The [kubelet](/docs/admin/kubelet/) uses liveness probes to know when to
+restart a Container. For example, liveness probes could catch a deadlock,
+where an application running, but unable to make progress. Restarting a
+Container in such a state can help to make the application more available
+despite bugs.
+
+The kubelet uses readiness probes to know when a Container is ready to start
+accepting traffic. A Pod is considered ready when all of its Containers are ready.
+One use of this signal is to control which Pods are used as backends for Services.
+When a Pod is not ready, it is removed from Service load balancers.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+{% include task-tutorial-prereqs.md %}
+
+{% endcapture %}
+
+{% capture steps %}
+
+### Defining a liveness command
+
+Many applications running for long periods of time eventually transition to
+broken states, and cannot recover except by being restarted. Kubernetes provides
+liveness probes to detect and remedy such situations.
+
+In this exercise, you create a Pod that runs a Container based on the
+`gcr.io/google_containers/busybox` image. Here is the configuration file for the Pod:
+
+{% include code.html language="yaml" file="exec-liveness.yaml" ghlink="/docs/tasks/configure-pod-container/exec-liveness.yaml" %}
+
+In the configuration file, you can see that the Pod has a single Container.
+The `livenessProbe` field specifies that the kubelet should perform a liveness
+probe every 5 seconds. The `initialDelaySeconds` field tells the kubelet that it
+should wait 5 second before performing the first probe. To perform a probe, the
+kubelet executes the command `cat /tmp/healthy` in the Container. If the
+command succeeds, it returns 0, and the kubelet considers the Container to be alive and
+healthy. If the command returns a non-zero value, the kubelet kills the Container
+and restarts it.
+
+When the Container starts, it executes this command:
+
+```shell
+/bin/sh -c "touch /tmp/healthy; sleep 30; rm -rf /tmp/healthy; sleep 600"
+```
+
+For the first 30 second of the Container's life, there is a `/tmp/healthy` file.
+So during the first 30 seconds, the command `cat /tmp/healthy` returns a success
```

Comment:
Starting with "So" after the last sentence is awkward, but I'm no editor, so grain of salt...